### PR TITLE
feat: add support for RSA-HSM and EC-HSM keys

### LINKED ISSUE #469: Support RSA-HSM keys
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
- Support `RSA-HSM` keys with the azure keyvault provider

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Adds support for `RSA-HSM` and `EC-HSM` keys

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #469 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: test/e2e/key_test.go [hunk 1]
```diff
@@ -128,4 +137,98 @@ var _ = Describe("When deploying SecretProviderClass CRD with keys", func() {
...
+		pod.WaitFor(pod.WaitForInput{
+			Getter:         kubeClient,
+			KubeconfigPath: kubeconfigPath,
+			Config:         config,
+			PodName:        p.Name,
+			Namespace:      ns.Name,
+		})
+
+		cmd := getPodExecCommand("cat /mnt/secrets-store/rsahsmkey1")
+		_, err = exec.KubectlExec(kubeconfigPath, p.Name, p.Namespace, strings.Split(cmd, " "))
```
Comment (sozercan) [inline test/e2e/key_test.go:184]: Should we check contents of the key?

Comment (aramase) [reply test/e2e/key_test.go:184]: @sozercan We can run some basic validation like ensuring the content is not nil and the `BEGIN PUBLIC KEY` and `END PUBLIC KEY` exists but I don't think we need the overhead of validating the key. WDYT?

Comment (sozercan) [reply test/e2e/key_test.go:184]: sgtm

Comment (aramase) [reply test/e2e/key_test.go]: done!

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**